src/main/java/com/neroreflex/plugins/Calc.java

```diff
@@ -18,65 +18,81 @@
  *
  * @author Nitti Gianluca
  */
 public final class Calc extends Trancio {
+
+    private class IrcWriter extends Writer{
+        private String chan;
+        private String buffer;
+
+        public IrcWriter(String chan){
+            this.chan = chan;
+            buffer = "";
+        }
+
+        @Override
+        public void write(char[] cbuf, int off, int len) throws IOException {
+            for(int i = off; i < len; i++)
+                buffer += cbuf[i];
+        }
+
+        @Override
+        public void flush() throws IOException {}
+
+        @Override
+        public void close() throws IOException {
+            if(buffer.length() != 0)
+                for(String s: buffer.split("[\\r\\n]+"))
+                    sendMessage(new Message(chan, s));
+            buffer = "";
+        }
+    }
+
     protected String onHelp() {
-        return "<number1> <operation> <number2> - where operation can either be +, - * or /";
+        return "Use !calc <expression> to evaluate an expression; supports integer and decimal numbers, parenthesis and +,-,*,/,^ binary operators. !calc verbose <expression> shows the steps done.";
     }
     
     @Override
     public final void onCall(Request req) {
         String user = req.GetUser(), channel = req.GetChannel();
         Vector<String> args = req.GetBasicParse();
-        
-        if(args.size() != 3)
-            this.sendMessage(new Message(channel, user + " Invalid arguments. Look at the plugin help to understand how to invoke cald"));
-        else{
-            double a;
-            double b;
-            try{
-                a = Double.parseDouble(args.get(0));
-                b = Double.parseDouble(args.get(2));
-            }catch(NumberFormatException e){
-                this.sendMessage(new Message(channel, user + " the given operands aren't numbers"));
-                return;
-            }
-            char op = args.get(1).charAt(0);
-            double r = 0;
-            switch(op){
-                case '+':
-                    r = a + b;
-                    break;
-                case '-':
-                    r = a - b;
-                    break;
-                case '*':
-                    r = a * b;
-                    break;
-                case '/':
-                    r = a / b;
-                    break;
-                default:
-                    sendMessage(new Message(channel, user + " unknown operator \"" + args.get(1) + "\" :(. Only +,-,*,/ are supported"));
-                    return;
+        String result;
+        IrcWriter w = new IrcWriter(user); //un eventuale log viene mandato in provato all'utente che ha richiesto il calcolo
+        try {
+            if(args.size() == 0){
+                result = "please specify an expression or a command (see help for further information)";
+            }else if(args.size() == 1){ //Se viene specificato 1 solo argomento, allora dev'essere un'espresione
```

Comment:
Il parsing fatto così non mi piace: non si può spezzare una espressione in più parti.

Il modo secondo me più adeguato e ciclare ogni argomento in args e fare un .compareTo(...) a possibili argomenti, tipo "verbose", "round=3", "integer_only=yes" e gli altri argomenti, cioè quelli successivi all'ultimo argomento valido vanno uniti in una sola stringa e poi evalutati.